internals/garbage-collector.rst

```diff
@@ -53,9 +53,24 @@ is needed to clean these reference cycles between objects once they become
 unreachable. This is the cyclic garbage collector, usually called just Garbage
 Collector (GC), even though reference counting is also a form of garbage collection.
 
+CPython contains two GC implementations starting in version 3.13. One implementation
+is used in the default build and relies on the global interpreter lock for
+thread-safety. The other implementation is used in the free-threaded build. Both
+implementations use the same basic algorithms, but operate on different data
+structures.
```

Comment:
This could use a list to make the difference between the two builds stand out more, and to mirror the two sections below.  Something like:
```suggestion
Starting in version 3.13, CPython contains two GC implementations starting in version 3.13:
* the default build implementation relies on the global interpreter lock for
thread-safety;
* the free-threaded build ...
 
Both implementations use the same basic algorithms, but operate on different data
structures.
```

Here you could also add link to `:term:`s like GC and GIL using intersphinx.

Reply:
I added a section at the end to summarize the differences between the two implementations. I was hesitant to add it here because we introduce a number of the concepts later on in the document.